src/engine.rs

```diff
@@ -1,18 +1,60 @@
 use anyhow::{anyhow, Result};
+use rust_embed::RustEmbed;
 use std::{
+    collections::HashSet,
     io::Cursor,
     path::PathBuf,
     time::{Duration, Instant},
 };
-use wasmtime::{Engine, Linker, Module, Store};
+use wasi_common::WasiCtx;
+use wasmtime::{Config, Engine, Linker, Module, Store};
 
 use crate::function_run_result::{
     FunctionOutput::{self, InvalidJsonOutput, JsonOutput},
     FunctionRunResult, InvalidOutput,
 };
 
+#[derive(RustEmbed)]
+#[folder = "providers/"]
+struct StandardProviders;
+
+fn import_modules(
+    module: &Module,
+    engine: Engine,
+    linker: &mut Linker<WasiCtx>,
+    mut store: &mut Store<WasiCtx>,
+) {
+    let imported_modules: HashSet<String> =
+        module.imports().map(|i| i.module().to_string()).collect();
+    imported_modules.iter().for_each(|imported_module| {
+        let mut imported_module = imported_module.to_owned();
+        imported_module.push_str(".wasm");
+
+        let imported_module_bytes = StandardProviders::get(&imported_module);
```

Comment:
Nit pick, but this feels a bit cleaner to me:

```suggestion
        let imported_module_bytes = StandardProviders::get(&format!("{imported_module}.wasm"));
```